integrations/zoom/events.go

```
@@ -0,0 +1,170 @@
+		return
+	}
+
+	// Retrieve all the relevant connections for this event.
+	ctx := r.Context()
+	cids, err := h.vars.FindConnectionIDs(ctx, desc.ID(), accountIDVar, accountID)
+	if err != nil {
+		h.logger.Error("failed to find connection IDs",
+			zap.String("account_id", accountID),
+			zap.Error(err))
+		common.HTTPError(w, http.StatusInternalServerError)
+		return
+	}
+
+	// Dispatch the event to all connections for potential asynchronous handling.
+	common.DispatchEvent(ctx, h.logger, h.dispatch, akEvent, cids)
+}
+
+// checkRequest validates the Zoom webhook HTTP request by verifying the content type,
+// required headers, and signature and returns the parsed JSON payload if valid.
+// Otherwise it returns nil with an HTTP error response.
+func (h handler) checkRequest(w http.ResponseWriter, r *http.Request) map[string]any {
+	l := h.logger.With(
+		zap.String("url_path", r.URL.Path),
+		zap.String("event_type", r.Header.Get("Zoom-Event")),
+	)
+
+	// Check the request's HTTP headers.
+	ct := r.Header.Get("Content-Type")
+	if !strings.HasPrefix(ct, "application/json") {
+		l.Warn("incoming event: unexpected content type", zap.String("content_type", ct))
+		common.HTTPError(w, http.StatusBadRequest)
+		return nil
+	}
+
+	signature := r.Header.Get("x-zm-signature")
+	timestamp := r.Header.Get("x-zm-request-timestamp")
+
+	if signature == "" || timestamp == "" {
+		l.Warn("incoming event: missing Zoom verification headers")
+		common.HTTPError(w, http.StatusBadRequest)
+		return nil
+	}
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		l.Error("incoming event: failed to read request body", zap.Error(err))
+		common.HTTPError(w, http.StatusInternalServerError)
+		return nil
+	}
+	r.Body = io.NopCloser(bytes.NewBuffer(body))
+
+	secret, err := h.signingSecret()
+	if err != nil {
+		l.Error("incoming event: signing secret not found", zap.Error(err))
+		common.HTTPError(w, http.StatusInternalServerError)
+		return nil
+	}
+
+	if secret == "" {
```

why is this check necessary when you check it in `signingSecret`?
I.e. if `secret == ""` it will never make it to line 125

---

Good question. Mario asked me about this too :)

`signingSecret()` (in Linear and here) currently supports only default OAuth. When we'll support private OAuth apps there:

- If the relevant connections don't contain a secret token, return an error, just like the existing check for default OAuth apps
- If there are no relevant connections, we should silently ignore the event, not report an error - so return `"", nil`.